src/mds/objects.py

```diff
@@ -226,6 +226,161 @@ async def create_object_for_id(
+                    "/programs/c/projects/a"
+                ],
+                "pet": "ferret",
+                "pet_age": 5,
+                "sport": "soccer"
+            },
+            "2": {
+                "message": "morning",
+                "_uploader_id": "102",
+                "_resource_paths": [
+                    "/programs/d",
+                    "/programs/e"
+                ],
+                "counts": [42, 42, 42],
+                "pet": "ferret",
+                "pet_age": 10,
+                "sport": "soccer"
+            },
+            "3": {
+                "message": "evening",
+                "_uploader_id": "103",
+                "_resource_paths": [
+                    "/programs/f/projects/a",
+                    "/admin"
+                ],
+                "counts": [1, 3, 5],
+                "pet": "ferret",
+                "pet_age": 15,
+                "sport": "basketball"
+            }
+        }
+
+    how do we design a filtering interface that allows the user to get all
+    objects having an authz string matching the pattern
+    "/programs/%/projects/%" at any index in its "_resource_paths" array? (%
+    has been used as the wildcard so far because that's what Postgres uses as
+    the wildcard for LIKE) In this case, the "1" and "3" objects should be
+    returned.
+
+    The filter syntax that was arrived at ending up following the syntax
+    specified by a [Node JS implementation](https://www.npmjs.com/package/json-api#filtering) of the [JSON:API
+    specification](https://jsonapi.org/).
+
+    The format for this syntax is filter=(field_name,operator,value), in which
+    the field_name is a json key without quotes, operator is one of :eq, :ne,
+    :gt, :gte, :lt, :lte, :like, :all, :any (see operators dict), and value is
+    a typed json value against which the operator is run.
+
+    Examples:
+
+        GET /objects?filter=(message,:eq,"morning") returns "2"
+        GET /objects?filter=(counts.1,:eq,3) returns "3"
+        GET /objects?filter=(pet_age,:lte,5) returns "0" and "1"
+        GET /objects?filter=(pet_age,:gt,5) returns "2" and "3"
+
+    Compound expressions are supported:
+
+        GET /objects?filter=(_resource_paths,:any,(,:like,"/programs/%/projects/%")) returns "1" and "3"
+        GET /objects?filter=(counts,:all,(,:eq,42)) returns "2"
+
+    Boolean expressions are also supported:
+
+        GET /objects?filter=(or,(_uploader_id,:eq,"101"),(_uploader_id,:eq,"102")) returns "1" and "2"
+        GET /objects?filter=(or,(and,(pet,:eq,"ferret"),(sport,:eq,"soccer")),(message,:eq,"hello")) returns "0", "1", and "2"
+    """
+
+    metadata_objects = await search_metadata_objects(
+        data=data, page=page, limit=limit, filter=filter
+    )
+
+    records = {}
+    if data and metadata_objects:
+        try:
+            endpoint_path = "/bulk/documents"
+            full_endpoint = config.INDEXING_SERVICE_ENDPOINT.rstrip("/") + endpoint_path
```

Comment:
https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin might be nicer than having to manually join the components properly

Reply:
When I'm testing locally, I like to have `INDEXING_SERVICE_ENDPOINT="https://john.planx-pla.net/index"`, but using `urljoin` here resulted in `full_endpoint` being set to `"https://john.planx-pla.net/bulk/documents"`.

Reply:
You're right! Sorry about that. I could've sworn that it could properly join a relative path, but I was mistaken. I think I'd suggested the same for @paulineribeyre without being cognizant of that issue. Sorry for the poor suggestion! I guess it's really only applicable for a base + path. Too bad it doesn't have a better `join()` operator for generic URL construction 🤔